plugins/inputs/logfile/tail/tail.go

```diff
@@ -225,6 +226,16 @@ func (tail *Tail) readLine() (string, error) {
 	tail.lk.Lock()
 	defer tail.lk.Unlock()
 
+	// do not allow line reading if the output pusher size is greater than configured
+	block, bufferSize, maxBufferSize := util.GetLogBlocker().Block()
+	for block {
```

Comment:
How will we handle this in the shutdown scenario for the agent? In the scenario that we are blocking indefinitely, at what point does this tailing goroutine terminate?

Reply:
Added a context with cancel from the background context, so it exits the chan on parent context cancel.